internal/git/gitsync_processor.go

```diff
@@ -601,7 +584,18 @@ func NewGitSyncProcessor(ctx context.Context, gitSync *v1alpha1.GitSync, kubeCli
 	}
 	localRepoPath := getLocalRepoPath(gitSync.GetName())
 	go func(repo *v1alpha1.RepositoryPath) {
-		r, err := cloneRepo(ctx, localRepoPath, repo)
+		globalConfig, err := controllerconfig.GetConfigManagerInstance().GetConfig()
+		if err != nil {
+			logger.Errorw("error getting  the  global config", "err", err)
+		}
+
+		gitCredentials := gitshared.FindCredByUrl(repo.RepoUrl, globalConfig)
```

Comment:
It should be allowed for a given repository that there are no credentials and so nothing in the ConfigMap. Need to account for that case.

Reply:
It should also be allowed that this prefix is in the ConfigMap but there are no credentials with it. Please make sure that you account for and test both of these cases.